### ISSUE TITLE
use capacitor for ionic archive

### ISSUE THREAD

**Issue by aabanaag** [bitrise-steplib/steps-ionic-archive#57]
It is possible to use capacitor instead of Cordova for the archive process?

Got this error but I can't find a way to use capacitor instead of cordova
```
[ERROR] Ionic doesn't support using Cordova with @ionic/react projects.
        
        We encourage you to try ⚡️ Capacitor ⚡️ (https://ion.link/capacitor)
        
ionic prepare command ionic "cordova" "prepare" "--no-build" failed, error: exit status 1
```

**Comment by non-binary** [comment]
Hi @aabanaag,

We are working on adding Capacitor support but it is currently not ready yet.

Please use a script step to do this.

**Comment by aabanaag** [comment]
Thank you for the update. But just to clarify so the step will be the same right? I still need to generate the build config then instead of ionic archive step I'll use the script to trigger `ionic cap sync ios` right?

**Comment by aabanaag** [comment]
If I simply trigger the command from the script I got this error

```
Running Ionic capacitor
> capacitor sync ios
[error] Capacitor could not find the web assets directory "/Users/vagrant/git/build".
    Please create it, and make sure it has an index.html file. You can change
    the path of this directory in capacitor.config.json.
    More info: https://capacitor.ionicframework.com/docs/basics/configuring-your-app
[ERROR] An error occurred while running subprocess capacitor.
        
        capacitor sync ios exited with exit code 1.
        
        Re-running this command with the --verbose flag may provide more information.
```
All I did was add a script step then add `ionic cap sync ios` command

**Comment by non-binary** [comment]
Hi @aabanaag,

I'm not super familiar with Capacitor or Ionic but can you please confirm tat your web assets directory is in there?

**Comment by aabanaag** [comment]
Yes all of the web assets are there. I'll try to figure it out since there are output paths that the original ionic archive returns which is also a requirement for the next steps. Thanks!

**Comment by mbecca** [comment]
> Yes all of the web assets are there. I'll try to figure it out since there are output paths that the original ionic archive returns which is also a requirement for the next steps. Thanks!

hello, did you find some alternative solution / workaround?

Regards,

M.

**Comment by aabanaag** [comment]
No, I ended up building it out separately instead.

**Comment by mbecca** [comment]
> No, I ended up building it out separately instead.


Could you share with me how the complete command was?  Thank you.

M.

**Comment by mbecca** [comment]
> Hi @aabanaag,
> 
> We are working on adding Capacitor support but it is currently not ready yet.
> 
> Please use a script step to do this.

Hello, do you know when it could be available?  Thank you.

M.

**Comment by non-binary** [comment]
Hi @mbecca,

We don't have a fixed timeline as of right now. All I am able to say is "soon". Sorry!

**Comment by aabanaag** [comment]
hey @mbecca ,

For the steps, just setup your app not as an ionic project but just an iOS project, then on the script step add this

```
#!/usr/bin/env bash
# fail if any commands fails
set -e
# debug log
set -x

npm install -g npm@6.14.4
npm install
npm run build --verbose
npx cap sync ios
```

then for cocoapods, set the source path to `./ios` also the `BITRISE_PROJECT_PATH` to `ios/App/App.xcworkspace`

**Comment by aabanaag** [comment]
Any update regarding the support for capacitor?

**Comment by dawidglowczyk** [comment]
@aabanaag I've got a problem with scheme setting in 'Xcode Archive & Export' step. I set it up to 'App' and every time I try to build I got the following error:
`Failed to open project: /Users/vagrant/git/ios/App/App.xcworkspace: could not get scheme (App) from path (/Users/vagrant/git/ios/App/App.xcworkspace): scheme App not found in App`

Could you share your settings for this step?

**Comment by non-binary** [comment]
Still nothing unfortunately but it is being investigated.

**Comment by SrinivasanTarget** [comment]
@aabanaag Can you please share your `bitrise.yml` file here if possible?

@luna-kx any updates on the support?

**Comment by luna-kx** [comment]
Nothing yet unfortunately

**Comment by marioshtika** [comment]
Any update on this?

**Comment by scottopolis** [comment]
We are also looking for Capacitor support

**Comment by SrinivasanTarget** [comment]
@scottopolis I don't see it's worth waiting for an update here as I see they don't have any plans yet. We switched to use `fastlane` and it works pretty well for us.

**Comment by scottopolis** [comment]
@SrinivasanTarget do you mind sharing your code for the fastlane step?

**Comment by s1rc** [comment]
@SrinivasanTarget I'm also interested in how you configured your Fastlasne for Capacitor.

**Comment by fjaeger** [comment]
Would be great to see official Capacitor support for Bitrise in combination with Ionic!

**Comment by nbadiganti** [comment]
Would be great to see support for capacitor bitrise builds.

**Comment by benfavre** [comment]
hello ?

**Comment by MeMeMax** [comment]
There is unfortunately pretty bad documentation. I can´t figure out in which order I have to add which steps for an Ionic Capacitor deployment. I don´t have a macbook and therefore I need to sign through a Workflow. But no chance with the current documentation.

**Comment by reynirf** [comment]
This is still an important issue.

**Comment by marioshtika** [comment]
still important to me

**Comment by dserrano-netquest** [comment]
still important, I do not want to pivot to another tool, please consider this topic a priority!!!

**Comment by marioshtika** [comment]
I know there is not a step yet for this on Bitrise, but they have created a sample project and a bitrise.yml file that will deploy it both on iOS and Android.

Take look at this link
https://github.com/bitrise-io/Bitrise-Capacitor-Sample/blob/main/bitrise.yml
and build your way up from that

**Comment by dserrano-netquest** [comment]
thank you very much, and thank you for linking it here, maybe other lost souls find this thread before finding the bitrise custom .yml catalog

**Comment by sindu** [comment]
Hi guys,
Exactly 4 years ago a qustion arose about using capacitor instead of Cordova for the archive process?
You have on several occasions answered "it will be soon available"...
Now, 4 (FOUR) years have passed and still nothing.
Please, make this high priority!!!

You are supporting Ionic apps - many developers are trying to move from cordova to capacitor...

**Comment by istx25** [comment]
# I no longer work at Bitrise. I am replying because I'll be unsubscribing to this thread after doing so. 

@sindu It is unlikely you'll get a response from Bitrise from this thread alone. If you wish to rebump this feature request for them to build it, reach out to their developer support. They may or may not prioritize it (as I said I'm no longer affiliated with Bitrise). With that said, this repository is licensed under MIT and is open source. If you'd like to see support for this feature and depend on it as you say you do, I recommend reading the documentation and building it to contribute it for others to also benefit from. https://github.com/bitrise-steplib/steps-ionic-archive?tab=readme-ov-file#-contributing.

Best of luck.

**Comment by mrsaleh** [comment]
4 years and still this issue is open ? unbelievable.

**Comment by dserrano-netquest** [comment]
Hi, to the new people reaching, the link posted by [marioshtika](https://github.com/marioshtika) looks like a good place to start. I have been using a modified version for my specs at work.

Repost of the link: https://github.com/bitrise-io/Bitrise-Capacitor-Sample/blob/main/bitrise.yml
